packages/ui/src/components/va-tabs/VaTab/VaTab.vue

```diff
@@ -39,111 +40,46 @@
+import { defineComponent } from 'vue'
 import VaIcon from '../../va-icon'
+import { useRouterLinkProps } from '../../../composables/useRouterLink'
+import { useTab } from '../hooks/useTab'
 
-import { TabsServiceKey, TabsService } from '../VaTabs.vue'
-
-type Context = {
-  tabsService: TabsService;
-  hasKeyboardFocus: boolean;
-  keyboardFocusListeners: Record<string, () => void>;
-}
-
-class TabProps {
-  color = prop<string>({ type: String, default: undefined })
-  icon = prop<string>({ type: String, default: null })
-  label = prop<string>({ type: String, default: null })
-  disabled = prop<boolean>({ type: Boolean })
-  name = prop<string | number>({ type: [String, Number] })
-  tag = prop<string>({ type: String, default: 'div' })
-}
-
-const TabPropsMixin = Vue.with(TabProps)
-
-@Options({
+const VaTab = defineComponent({
   name: 'VaTab',
   components: { VaIcon },
   emits: ['click', 'keydown-enter', 'focus'],
-})
-export default class VaTab extends mixins(
-  RouterLinkMixin,
-  TabPropsMixin,
-  ColorMixin,
-) {
-  id: string | number | null = null
-  isActive = false
-  rightSidePosition = 0
-  leftSidePosition = 0
-  hoverState = false
 
-  context: Context = setup(() => {
-    const tabsService = inject(TabsServiceKey) as TabsService
-
-    const { hasKeyboardFocus, keyboardFocusListeners } = useKeyboardOnlyFocus()
-
-    return {
-      tabsService,
-      hasKeyboardFocus,
-      keyboardFocusListeners,
-    }
-  })
-
-  get classComputed () {
-    return {
-      'va-tab--disabled': this.$props.disabled,
-    }
-  }
-
-  get computedStyle () {
-    return {
-      color: this.context.hasKeyboardFocus || this.hoverState || this.isActive ? this.colorComputed : 'inherit',
-    }
-  }
-
-  get tabIndexComputed () {
-    return (this.$props.disabled || this.context.tabsService.disabled) ? -1 : 0
-  }
-
-  onTabClick () {
-    this.context.tabsService.tabClick(this)
-    this.$emit('click')
-  }
-
-  onTabKeydown () {
-    this.context.tabsService.tabPressEnter(this)
-    this.$emit('keydown-enter')
-  }
-
-  onFocus () {
-    if (this.context.hasKeyboardFocus) {
-      this.context.tabsService.tabFocus(this)
-    }
-    this.$emit('focus')
-  }
-
-  updateHoverState (isHover: boolean) {
-    this.hoverState = isHover
-  }
-
-  updateSidePositions () {
-    this.rightSidePosition = this.$el.offsetLeft + this.$el.offsetWidth
-    this.leftSidePosition = this.$el.offsetLeft
-  }
-
-  beforeMount () {
-    this.context.tabsService.register(this)
-  }
+  props: {
+    ...useRouterLinkProps,
+    color: {
+      type: String,
+      default: '',
+    },
+    icon: {
+      type: String,
+      default: '',
+    },
+    label: {
+      type: String,
+      default: '',
+    },
+    disabled: {
+      type: Boolean,
+    },
+    name: {
+      type: [String, Number],
+    },
+    tag: {
+      type: String,
+      default: 'div',
+    },
+  },
+
+  setup: useTab,
```

Comment:
Same problem here (as for tabs). All logic disappeared from component etc.

Reply:
Fixed.